data/rest-data.go

```diff
@@ -148,46 +132,36 @@ func (r *RestData) checkFile(filename string) (filepath string) {
 	return filepath
 }
 
-func (r *RestData) getWorkflowFiles() error {
-
-	//Only subdirectories are not allowed in the .github/workflows directory, so no need to recurse
-	endpoint := fmt.Sprintf("%s/repos/%s/%s/contents/.github/workflows", APIBase, r.owner, r.repo)
-	responseData, err := r.MakeApiCall(endpoint, true)
-	if err != nil {
-		r.Config.Logger.Error(fmt.Sprintf("Error calling github to retrive workflow files list: %s", err.Error()))
-		return err
+func (r *RestData) GetDirectoryContent(path string) (dirContent []*github.RepositoryContent, err error) {
+	workflowsDir, exists := r.contents.GetSubdirContentByPath(path)
+	if !exists {
+		return nil, fmt.Errorf("content not found at %s", path)
 	}
 
-	var workflowFileList []DirContents
-	err = json.Unmarshal(responseData, &workflowFileList)
-	if err != nil {
-		r.Config.Logger.Error(fmt.Sprintf("Error unmarshalling json response for workflow files list: %s", err.Error()))
-		return err
-	}
-
-	//For each file, listed we need to get it and put it in a format the action parser can use
-	var dirFiles = make([]DirFile, len(workflowFileList))
-	for i, workflowFile := range workflowFileList {
-
-		response, err := r.MakeApiCall(workflowFile.URL, true)
-		if err != nil {
-			r.Config.Logger.Error(fmt.Sprintf("Could not get workflow file data from github, error: %s", err.Error()))
-			return err
+	for _, file := range workflowsDir.Content {
+		if file.GetType() != "file" {
+			continue
 		}
 
-		var dirFile DirFile
-		err = json.Unmarshal(response, &dirFile)
+		content, err := r.getSourceFile(r.owner, r.repo, file.GetPath())
 		if err != nil {
-			r.Config.Logger.Error(fmt.Sprintf("Could not Unmarshal json response for file data, error: %s", err.Error()))
-			return err
+			return nil, fmt.Errorf("failed to fetch workflow file %s: %s", file.GetPath(), err.Error())
 		}
-
-		dirFiles[i] = dirFile
+		dirContent = append(dirContent, content)
 	}
 
-	r.Contents.WorkFlows = dirFiles
+	return dirContent, nil
+}
 
-	return err
+func (r *RestData) GetFileContent(path string) (content *github.RepositoryContent, err error) {
+	content, err = r.getSourceFile(r.owner, r.repo, path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve file content for %s: %w", path, err)
+	}
+	if content == nil {
+		return nil, fmt.Errorf("file not found at %s", path)
```

Comment:
❓ should this be `file not found` or `file has no content` or `file empty`? Or that is a length check like you have below?

Reply:
Nil content could be returned if the path was a directory... that may be the only reason a nil would be returned, though I'm not sure. Regarding length, I think the caller should decide whether zero length content equates to an error/failure.